src/main/java/frc/robot/Robot.java

```diff
@@ -56,11 +56,11 @@ public void disabledPeriodic() {}
   /** This autonomous runs the autonomous command selected by your {@link RobotContainer} class. */
   @Override
   public void autonomousInit() {
-    m_autonomousCommand = m_robotContainer.getAutonomousCommand();
+    Command command = RobotContainer.getAutonomousCommand();
 
```

Comment:
"autonomousCommand" was more descriptive.  Also, you have put the command in a local variable which leaves scope when autonomousInit() exits.  Therefore there is no place in the code where m_autonomousCommand is set, thus it won't be canceled in teleopInit().